frame/support/src/dispatch.rs

```diff
@@ -1485,6 +1597,35 @@ macro_rules! decl_module {
 		}
 	};
 
+	(@impl_on_idle
+		{ $system:ident }
+		$module:ident<$trait_instance:ident: $trait_name:ident$(<I>, $instance:ident: $instantiable:path)?>;
+		{ $( $other_where_bounds:tt )* }
+		fn on_idle($param1:ident : $param1_ty:ty, $param2:ident: $param2_ty:ty) -> $return:ty { $( $impl:tt )* }
+	) => {
+		impl<$trait_instance: $system::Config + $trait_name$(<I>, $instance: $instantiable)?>
+			$crate::traits::OnIdle<<$trait_instance as $system::Config>::BlockNumber>
+			for $module<$trait_instance$(, $instance)?> where $( $other_where_bounds )*
+		{
+			fn on_idle($param1: $param1_ty, $param2: $param2_ty) -> $return {
+				$crate::sp_tracing::enter_span!($crate::sp_tracing::trace_span!("on_idle"));
```

Comment:
the pallet macro doesn't implement these trace. I think you should put this span also in the OnIdle implementation in pallet macro.

And we should probably do same for other hooks.

Reply:
ya, I see what you are saying, will create an issue for this :)

Reply:
https://github.com/paritytech/substrate/issues/8303